service/tbc/crawler.go

```diff
@@ -1215,20 +1220,379 @@ func (s *Server) TxIndexer(ctx context.Context, endHash *chainhash.Hash) error {
 	return fmt.Errorf("invalid direction: %v", direction)
 }
 
+func processKeystones(blockHash *chainhash.Hash, txs []*btcutil.Tx, kssCache map[chainhash.Hash]tbcd.Keystone) error {
+	for _, tx := range txs {
+		if blockchain.IsCoinBase(tx) {
+			// Skip coinbase inputs
+			continue
+		}
+
+		for _, txOut := range tx.MsgTx().TxOut {
+			aPoPTx, err := pop.ParseTransactionL2FromOpReturn(txOut.PkScript)
+			if err != nil {
+				// log.Tracef("error parsing tx l2: %s", err)
+				continue
+			}
+			if _, ok := kssCache[*aPoPTx.L2Keystone.Hash()]; ok {
+				// Multiple keystones may exist in block, only store first
+				continue
```

Comment:
I don't think we should continue here; there may be a case where another keystone is found

Reply:
This is subtle but correct. We only care about the very first UNIQUE keystone.

See slack conversation for the explanation.

Reply:
![a](https://github.com/user-attachments/assets/b8c98e21-294a-4470-a61d-399d9494ca3c)